src/shared/metadata/cgroup_path_resolver.cc

```diff
@@ -27,12 +27,13 @@
 #include "src/common/fs/fs_wrapper.h"
 #include "src/shared/metadata/cgroup_path_resolver.h"
 
-DEFINE_bool(force_cgroup2_mode, true, "Flag to force assume cgroup2 fs for testing purposes");
+DEFINE_bool(force_cgroup2_mode, false, "Flag to force assume cgroup2 fs for testing purposes");
```

Comment:
I would add add `test_only` somewhere in the name of this flag.

Reply:
Additionally, now that we have a more robust cgroup resolver, can we get rid of this flag, and just rely on different testdata to make sure we cover {cgroup1 only, cgroup2 only, cgroup1+cgroup2 w/ cgroup1 failing, cgroup1+cgroup2 w/ cgroup1 succeeding}.

Reply:
Here are the complications with covering those cases:
- CGroupPathResolver (newer method) is [template based](https://github.com/pixie-io/pixie/blob/c6e18a9817c65d9ad5e350f49612a888266a74c3/src/shared/metadata/cgroup_path_resolver_test.cc#L35-L39) -- meaning it doesn't read the filesystem at all. This prevents us from testing against a directory of "mocked" cgroup data and simulating the failure seen in the end user's cluster.
- The end user's issue happened when the cgroup v1 base path existed, but the PEM was running within a v2 cgroup. This is challenging to simulate with the template based CGroupPathResolver approach because we don't have test coverage for the [function containing](https://github.com/pixie-io/pixie/blob/c6e18a9817c65d9ad5e350f49612a888266a74c3/src/shared/metadata/cgroup_path_resolver.cc#L140-L144) the bug.
- LegacyCGroupPathResolver operates with real directories under test. This complicates testing the cgroup v2 logic because we check the [filesystem type](https://github.com/pixie-io/pixie/blob/c6e18a9817c65d9ad5e350f49612a888266a74c3/src/shared/metadata/cgroup_path_resolver.cc#L53-L55), which is always false for a `testdata` directory.

In order to achieve the ideal test coverage, the CGroupPathResolver needs test coverage for [AutoDiscoverCGroupTemplate](https://github.com/pixie-io/pixie/blob/c6e18a9817c65d9ad5e350f49612a888266a74c3/src/shared/metadata/cgroup_path_resolver.cc#L139) and for the legacy path resolver needs to be able to pass the [cgroup v2 filesystem check](https://github.com/pixie-io/pixie/blob/c6e18a9817c65d9ad5e350f49612a888266a74c3/src/shared/metadata/cgroup_path_resolver.cc#L53-L56). The latter is what I was trying to accomplish with an [overlayfs](https://github.com/pixie-io/pixie/pull/1981/files#diff-271b2b75cd10a1080667a372213c09a1f3b963cba201a6450fcd6db8122796d1R351-R356) (unfortunately that doesn't work).

I'm going to spend more time thinking about how to address those problems.

Reply:
After thinking about this more, I don't think there is an easy way to test this without rearchitecting the code quite a bit. My most recent research was around testing the  `AutoDiscoverCGroupTemplate` function. I was hoping I could recreate the bug with fake cgroup v1 files and a bind mount to `/sys/fs/cgroup` (or some subdirectory) in order to have that function fail. Bazel unfortunately does not like a cgroup filesystem bind mount, so this isn't viable.

I'm currently thinking that the test investment right now might not be worth holding up this fix. What do you think about putting a TODO here explaining the test coverage we wish we'd have?